gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -399,7 +401,16 @@ def self.build_sig(hook_mod, method_name, original_method, current_declaration,
   end
 
   def self.unwrap_method(mod, signature, original_method)
-    maybe_wrapped_method = CallValidation.wrap_method_if_needed(mod, signature, original_method)
+    method_name = original_method.name
+    maybe_wrapped_method = signature.method.owner.instance_method(method_name)
+    file_path, _line = maybe_wrapped_method.source_location
+
+    # If Sorbet's wrapper was overridden, we don't want to try to unwrap the registered signature
+    unless !@was_ever_final_names[method_name] && original_method != maybe_wrapped_method &&
+           file_path != WRAPPER_METHOD_SOURCE_FILE
+      maybe_wrapped_method = CallValidation.wrap_method_if_needed(mod, signature, original_method)
+    end
```

Comment:
I believe the first wrap always happens in `_methods.rb` inside `_on_method_added` and the one in call validation only happens when unwrapping.

I added two extra tests for abstract so that we can be sure. If overriding an abstract method, we don't unwrap, which is what we'd expect. And in the other way, when overriding a method with an abstract one, then we're still unwrapping and going up the hierarchy chain.